freqtrade/exchange/bitget.py

```diff
@@ -0,0 +1,213 @@
+                f"params：{pair}, {timeframe}, {since_ms}, {candle_type}, "
+                f"can't get_historic_ohlcv: {e}"
+            )
+            raise
+
+    async def _estimate_current_candle(
+        self, pair: str, timeframe: str, start_time: int
+    ) -> list[float]:
+        timeframe_map: dict[str, tuple[str, int]] = {
+            "5m": ("1m", 5),
+            "15m": ("3m", 5),
+            "30m": ("5m", 6),
+            "1h": ("15m", 4),
+            "4h": ("1h", 4),
+            "6h": ("1h", 6),
+            "12h": ("3h", 4),
+            "1d": ("6h", 4),
+            "1w": ("1d", 7),
+            "1m": ("1w", 4),
+        }
+
+        if timeframe not in timeframe_map:
+            raise Exception(f"{timeframe} not in timeframe list")
+
+        smaller_tf, max_candles = timeframe_map[timeframe]
+
+        try:
+            smaller_candles = await self._api_async.fetch_ohlcv(
+                pair, smaller_tf, start_time, max_candles
+            )
+
+            if not smaller_candles:
+                logger.warning(f"can't get  {smaller_tf} candle data for {pair}")
+                return []
+            open_price = smaller_candles[0][1]
+            high_price = max(candle[2] for candle in smaller_candles)
+            low_price = min(candle[3] for candle in smaller_candles)
+            close_price = smaller_candles[-1][4]
+            volume = sum(candle[5] for candle in smaller_candles)
+            return [start_time, open_price, high_price, low_price, close_price, volume]
+
+        except Exception as e:
+            raise Exception(f"can't for {pair} get candle of {timeframe}: {e}")
+
+    def ohlcv_candle_limit(
+        self, timeframe: str, candle_type: CandleType, since_ms: int | None = None
+    ) -> int:
+        if candle_type == CandleType.FUNDING_RATE:
+            return 100  # Bitget seems to limit funding rate data to 100 entries
+        if candle_type == CandleType.MARK:
+            return 200
+        # For other candle types, use the default or previously defined limit
+        return super().ohlcv_candle_limit(timeframe, candle_type, since_ms)
+
+    def get_funding_fees(
+        self, pair: str, amount: float, is_short: bool, open_date: datetime
+    ) -> float:
+        if self.trading_mode == TradingMode.FUTURES:
+            try:
+                return self._fetch_and_calculate_funding_fees(pair, amount, is_short, open_date)
+            except ExchangeError:
+                logger.warning(f"Could not update funding fees for {pair}.")
+        return 0.0
+
+    def get_max_pair_stake_amount(self, pair: str, price: float, leverage: float = 1.0) -> float:
+        if self.trading_mode == TradingMode.SPOT:
+            return float("inf")  # Not actually inf, but this probably won't matter for SPOT
+
+        if pair not in self._leverage_tiers:
+            return 1.0
+
+        pair_tiers = self._leverage_tiers[pair]
+        return pair_tiers[-1]["maxNotional"] / leverage
```

Comment:
I'm ... not convinced this is correct (the whole method).

It's disabling spot limits completely  - though the exchange provides them just fine (which is definitely wrong) - so "parent" should work perfectly fine.
Also for non-spot - i'm not sure i understand what makes this different to the parent method. It's simpler, and ignores a lot more stuff - but that doesn't make it correct.